operator/pkg/operator/injector/injector.go

```diff
@@ -252,10 +258,32 @@ func (s *SidecarInjectField) OverlaySidecar(a Annotations, ao *AnnotationOverlay
 			}
 		}
 	}
+
 	s.SidecarVolumeMount.Name = s.SidecarVolume.Name
 	s.ConfigmapVolumeMount.Name = s.ConfigmapVolume.Name
 	s.Initcontainer.VolumeMounts = []corev1.VolumeMount{s.SidecarVolumeMount}
 
+	// add requests and limits to initcontainer
+	if limitsStr != "nil" {
+		limits := make(corev1.ResourceList)
+		err := json.Unmarshal([]byte(limitsStr), &limits)
+		if err != nil {
+			log.Error(err, "unmarshal limitsStr error")
+			return false
+		}
+		s.Initcontainer.Resources.Limits = limits
```

Comment:
Would you please validate the string by a checker?

Reply:
OK, get it.